Swift/T Release 1.5.0

j-woz opened the pull request:
Simple version number updates to new downloadable packages

j-woz committed:
Swift/T Release 1.5.0

j-woz committed:
Swift/T Release 1.5.0

j-woz committed:
Merge branch 'develop' of github.com:spack/spack into develop

srekolam committed:
Enable image_support flag to fix the issues with hipMalloc3D,hipMallocPitch() (#23620)

jrood-nrel committed:
Add -DSTK_NO_BOOST_STACKTRACE to compile line in trilinos if +stk and on a Mac. (#23625)

ax3l committed:
openPMD-api: 0.13.4 (#23619)

Add the latest release.

eugeneswalker committed:
veloc: add v1.3 and v1.2 (#23626)

eugeneswalker committed:
py-warpx: new version: 21.05 (#23628)

frankwillmore committed:
cc: change mode to ccld for loopopt edit (#23482)

For configure (e.g. for hdf5) to pass, this option needs to be pulled out when invoked in ccld mode.

I thought it had fixed the issue but I still saw it after that. After some digging, my guess is that I was able
to get hdf5 to build with ifort instead of ifx. Lot of overlapping changes occurring at the time, as it were.
There are still outstanding issues building hdf5 with ifx, and Intel is looking into what appears to be a
compiler bug, but this manifests during build and is likely a separate issue.

I have verified that the making the edit in 'ccld' mode removes the -loopopt=0 and enables hdf5 to pass
configure. It should be fine to make the edit in 'ld' mode as well, but I have not tested that and didn't
include an -or- condition for it.

alalazo committed:
spec: simplify __str__ implementation (#23593)

The implementation for __str__ has been simplified to traverse the spec directly, 
and doesn't call anymore the flat_dependencies method. Dead code has been 
removed.

sebrowne committed:
netcdf-c: refactor fsync variant (#23477)

Simplify logic by just enabling or disabling fsync as user specified
(default to off currently).  Also remove the 4.1 version check, since
that version isn't actually supported in here.

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

natshineman committed:
Update mvapich2 to 2.3.6 (#23574)

Co-authored-by: Nat Shineman <shineman.5@buckeyemail.osu.edu>

vmiheer committed:
emacs: add support for native compilation on master (#23493)

* Emacs native compilation on master

* emacs: Mark tools correctly as build dependency

jacobmerson committed:
model-traits: add new package (#23589)

alalazo committed:
build tests: put an upper bound on the version of GCC being used (#23630)

pmccormick committed:
legion: bug fix for flecsi use cases. (#23624)

adamjstewart committed:
py-rarfile: add new package (#23622)

michaelkuhn committed:
emacs: add texinfo build dependency on master (#23631)

It seems that building unreleased versions requires makeinfo, which is
part of texinfo.

alalazo commented on var/spack/repos/builtin/packages/adlbx/package.py:
```diff
@@ -11,14 +11,16 @@ class Adlbx(AutotoolsPackage):
     git      = "https://github.com/swift-lang/swift-t.git"
 
     version('master', branch='master')
+    version('1.0.0', sha256='9e33cbd3cbdd3d85eb81bcc7e619e0a8')
```
We need to use sha256, can you update?

iarspider committed:
root: Add variants: dcache and oracle (#23441)

becker33 committed:
Separable module configurations (#22588)

Currently, module configurations are inconsistent because modulefiles are generated with the configs for the active environment, but are shared among all environments (and spack outside any environment).

This PR fixes that by allowing Spack environments (or other spack config scopes) to define additional sets of modules to generate. Each set of modules can enable either lmod or tcl modules, and contains all of the previously available module configuration. The user defines the name of each module set -- the set configured in Spack by default is named "default", and is the one returned by module manipulation commands in the absence of user intervention.

As part of this change, the module roots configuration moved from the `config` section to inside each module configuration.

Additionally, it adds a feature that the modulefiles for an environment can be configured to be relative to an environment view rather than the underlying prefix. This will not be enabled by default, as it should only be enabled within an environment and for non-default views constructed with separate projections per-spec.

TODO:
- [x] code changes to support multiple module sets
- [x] code changes to support modules relative to a view
- [x] Tests for multiple module configurations
- [x] Tests for modules relative to a view
- [x] Backwards compatibility for module roots from config section
- [x] Backwards compatibility for default module set without the name specified
- [x] Tests for backwards compatibility

haampie committed:
Bump coreutils (#23633)

michaelkuhn committed:
zstd: add 1.5.0 (#23646)

j-woz committed:
Merge branch 'develop' of github.com:j-woz/spack into develop
